test/functorch/test_aotdispatch.py

```diff
@@ -2062,6 +2063,197 @@ def f(a):
 
         self.assertEqual(inp_ref.grad, inp_test.grad)
 
+    def test_buffer_copied_in_graph(self):
+        class MyModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.register_buffer("buf", torch.zeros(1))
+                self.w1 = torch.nn.Parameter(torch.zeros(1))
+                self.w2 = torch.nn.Parameter(torch.zeros(1))
+
+            def forward(self, x):
+                self.buf.add_(1)
+                return (self.w1 * x * self.w2).sum() + self.buf.sum()
+
+        model_for_eager = MyModel()
+        model_for_compile = copy.deepcopy(model_for_eager)
+
+        eager_version_counters = [buffer._version for _, buffer in model_for_eager.named_buffers()]
+        compile_version_counters = [buffer._version for _, buffer in model_for_compile.named_buffers()]
+
+        print(eager_version_counters, compile_version_counters)
+
+        eager_version_counters = [buffer._version for _, buffer in model_for_eager.named_buffers()]
+        compile_version_counters = [buffer._version for _, buffer in model_for_compile.named_buffers()]
+
+        fw_graph_cell = [None]
+        compiled_f = aot_module(
+            model_for_compile,
+            fw_compiler=make_boxed_compiler(partial(extract_graph, graph_cell=fw_graph_cell)),
+            bw_compiler=nop,
+            keep_inference_input_mutations=True,
+        )
+        inp_ref = torch.ones(1, requires_grad=True)
+        inp_test = torch.ones(1, requires_grad=True)
+
+        out_ref = model_for_eager(inp_ref.clone())
+        out_test = compiled_f(inp_test.clone())
+
+        eager_version_counters = [buffer._version for _, buffer in model_for_eager.named_buffers()]
+        compile_version_counters = [buffer._version for _, buffer in model_for_compile.named_buffers()]
```

Comment:
mentioned offline - this test might be easier to do with inductor, since (as we learned haha) aot_eager will double-count version counter bumps (since aot_eager will just run each node under an fx interpreter, including `copy_()`, and autograd will see it).

Maybe we can "fix" this in aot_eager by having that backend run under inference_mode, although that seems like it should be its own separate fix.